docs/grammar/SolidityParser.g4

```diff
@@ -169,23 +170,14 @@ locals[
 	(
 		{!$visibilitySet}? visibility {$visibilitySet = true;}
 		| {!$mutabilitySet}? stateMutability {$mutabilitySet = true;}
-		| modifierInvocation
+		| {!$free}? modifierInvocation
```

Comment:
ah ok.. I see.. so the `free` parameter is used here.. ah and based on this parameter you basically define the needed tokens?

Reply:
Yes.

Reply:
> but shouldn't it also work with only `{!$free}`

Not sure what you mean. It works only with `{!$free}`. There's no other variable used here.

Reply:
The question is whether it could be ``{!$free} modifierInvocation`` instead of ``{!$free}? modifierInvocation``, interpreting the ``?`` as ``$free`` may or may not have a value yet, if I understand it correctly. If I recall correctly (I haven't looked into the details of antlr syntax myself for a while), that's not how this ``?`` is read and it's rather *always* part of conditional rules, i.e. it's always ``{...}? ...`` if the rule is dependent on some boolean condition.

Reply:
[resolving this based on the assumption that that answered the question]